docs/rfc/active/graph_ql_frontend/queries.md

```diff
@@ -0,0 +1,643 @@
+
+    uri: String
+
+    platformNativeType: PlatformNativeType
+
+    tags: [String]!
+
+    properties: [PropertyTuple]
+
+    createdTime: Long!
+
+    modifiedTime: Long!
+
+    ownership: Ownership
+}
+
+"""
+Represents Ownership
+"""
+type Ownership {
+
+    owners: [Owner]
+
+    lastModified: Long!
+}
+
+"""
+Represents an Owner
+"""
+type Owner {
+    """
+     The fully-resolved owner
+    """
+    owner: CorpUser!
+
+    """
+     The type of the ownership
+    """
+    type: OwnershipType
+
+    """
+     Source information for the ownership
+    """
+    source: OwnershipSource
+}
+```
+
+Notice that the Dataset's ``Ownership`` aspect includes a nested ``Owner`` field that references a ``CorpUser`` type. In the GMS model, this is represented as a foreign-key relationship (urn). In GraphQL, we include the *resolved* relationship, allowing the client to easily retrieve information about a Dataset's owners.
+
+To support traversal of this relationship, we additionally include a ``CorpUser`` type:
+
+```graphql
+"""
+Represents the CorpUser GMA Entity
+"""
+type CorpUser {
+
+    urn: String!
+
+    username: String!
+
+    info: CorpUserInfo
+
+    editableInfo: CorpUserEditableInfo
+}
+```
+
+#### 2. **Extend the 'Query' type**: 
+
+GraphQL defines a top-level 'Query' type that serves as the entry point for reads against the graph. This is extended to support querying the new entity type.
+
+```graphql
+type Query {
+    dataset(urn: String!): Dataset # Add this! 
+    datasets(urn: [String]!): [Dataset] # Or if batch support required, add this! 
+}
+```
+
+#### 3. **Define & Register DataLoaders**
+This is illustrated in the previous section. It involves extending `DataLoader` and registering the loader in the `DataLoaderRegistry`.
+
+#### 4. **Define & Register DataFetcher (Data Resolver)**
+This is illustrated in the previous section. It involves implementing the DataFetcher interface and attaching it to fields in the graph.
+
+#### 5. **Query your new entity**
+Deploy & start issuing queries against the graph. 
+
+```graphql
+// Input
+query datasets($urn: String!) { 
+    dataset(urn: $urn) { 
+        urn 
+        ownership {
+            owners {
+                owner {
+                    username
+                }
+            }
+        }
+    }
+}
+
+// Output 
+{
+    "data": {
+        "dataset": {
+            "urn": "urn:li:dataset:(urn:li:dataPlatform:foo,bar,PROD)",
+            "ownership": {
+                "owners": [
+                    {
+                        "owner": {
+                            "username": "james joyce",
+                        }
+                    }
+                ]
+            }
+        }
+    }
+}
+```
+
+**Disclaimer**
+
+It is the intention that the GraphQL type system be **auto-generated** based on the PDL models declared at the GMA layer. This means that the frontend schema should need not be maintained separately from the GMA it is derived from. 
```

Comment:
@jplaisted : assumption of auto-generation from PDL -> GQL is described here.

Reply:
Got it, thanks for the call out! :)